gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -214,24 +258,41 @@ def compute_inliers(
         projection_directions = self.__sample_projection_directions(combined_measurements)
 
         # Convert to measurements: map indexes to symbols.
-        dummy_noise_model = gtsam.noiseModel.Isotropic.Sigma(3, 1e-2)  # MFAS does not use this.
-        w_i1Ui2_measurements = self._binary_measurements_from_dict(w_i2Ui1_dict, w_iUj_dict_tracks, dummy_noise_model)
-
-        # Compute outlier weights using MFAS.
-        # TODO(ayush): parallelize this step.
-        outlier_weights: List[Dict[Tuple[int, int], float]] = []
-        for direction in projection_directions:
-            mfas_instance = MFAS(w_i1Ui2_measurements, direction)
-            outlier_weights.append(mfas_instance.computeOutlierWeights())
+        w_i1Ui2_measurements = self._binary_measurements_from_dict(w_i2Ui1_dict, w_iUj_dict_tracks, DUMMY_NOISE_MODEL)
+
+        # TODO(johnwlambert): Consider getting global client and scattering large data.
+        future_w_i2Ui1_dict = w_i2Ui1_dict
+        future_w_iUj_dict_tracks = w_iUj_dict_tracks
+
+        # Loop through tracks and and generate delayed MFAS tasks.
+        batch_size = int(np.ceil(len(projection_directions) / self._max_delayed_calls))
+        batched_outlier_weights: List[Any] = []
+        if batch_size == 1:
+            logger.info("BATCH SIZE 1")
+            for direction in projection_directions:
+                batched_outlier_weights.append(
+                    dask.delayed(MFASWrapper)(MFAS, future_w_i2Ui1_dict, future_w_iUj_dict_tracks, [direction])
+                )
+        else:
+            for j in range(0, len(projection_directions), batch_size):
+                batched_outlier_weights.append(
+                    dask.delayed(MFASWrapper)(
+                        MFAS, future_w_i2Ui1_dict, future_w_iUj_dict_tracks, projection_directions[j : j + batch_size]
+                    )
+                )
+
+        # Compute outlier weights in parallel.
+        batched_outlier_weights = dask.compute(*batched_outlier_weights)
```

Comment:
This might be a dask anti-pattern. https://docs.dask.org/en/latest/delayed-best-practices.html

We should not be calling compute repeatedly. We to have a function which unzips and chains the batch results

Reply:
Could you elaborate a little more? I'm only calling compute once.

Reply:
Could you remove this since we agreed that the architecture is good?

Reply:
Removed.